### PR TITLE
Addresses issue #30.

### PR DESCRIPTION
**Pull request by MarcusBarnes**
Addresses PHP notice generated if user supplies invalid IP range as input to reaction. See issue #30.

### CODE REVIEW

File: plugins/islandora_context_reaction_restrict_by_ip.inc [hunk 1]
```diff
@@ -111,7 +111,7 @@ class IslandoraContextReactionRestrictByIp extends context_reaction {
       if (!strlen($range)) {
         continue;
       }
-      list($low, $high) = explode(':', $range);
+      list($low, $high) = array_pad(explode(':', $range, 2), 2, NULL);
```
Comment (mjordan) [inline plugins/islandora_context_reaction_restrict_by_ip.inc:114]: @MarcusBarnes the addition of `array_pad()` does get rid of the notice when given a single IP address without a `:`, but `NULL` is being assigned to `$high`, which is then being passed into `ip2long()`. I think we need to add logic to test whether `$high` is `NULL` and if so, determine if access from `$low` as a single IP address, not as the start of a range, is allowed. We'd also need to update the help text in the IP range field to indicate that a single IP is allowed.

If you're OK with the additional logic and help text, I can add it, or if you want to, go ahead.

Comment (MarcusBarnes) [reply plugins/islandora_context_reaction_restrict_by_ip.inc:114]: After looking at the code and using the plugin a bit more, I felt that the current expectation that users provide single IPs using the range notation keeps the lines in the input field more consistent.  So I only fixed the notice, rather than adding additional logic to handle single IP addresses.

I suggest that we merge this fix to the notice, then determine if it's really desirable to mix up single IPs and IP ranges in the input form of the restrict by IP reaction plugin.   Thank you for considering.

Comment (mjordan) [reply plugins/islandora_context_reaction_restrict_by_ip.inc:114]: OK, I'll merge. Do you want to open a new ticket for the discussion about how to represent single IP addresses?

Comment (MarcusBarnes) [reply plugins/islandora_context_reaction_restrict_by_ip.inc:114]: Thank you!  After working with the plugin a little more, I've come to like the consistency of using the IP range notation even for single IP addresses.   Maybe create a question ticket in case others feel strongly about not using the range notation for single IP addresses?

Comment (mjordan) [reply plugins/islandora_context_reaction_restrict_by_ip.inc:114]: When you say "using the IP range notation even for single IP addresses" do you mean something like:

```
123.45.12.234:123.45.12.234
```

Personally, I see no problem in mixing ranges and single addresses in one form field as long as the exact notation is documented clearly, like:

```
142.58.0.0:142.58.79.255
142.58.88.0:142.58.101.26
123.45.12.234
```

But, I'd be happy to discuss further or to open a ticket to get wider input.

Comment (MarcusBarnes) [reply plugins/islandora_context_reaction_restrict_by_ip.inc:114]: @mjordan Correct - when I wrote "using the IP range notation even for single IP addresses" I meant using notation like:

``123.45.12.234:123.45.12.234``

Comment (mjordan) [reply plugins/islandora_context_reaction_restrict_by_ip.inc:114]: @MarcusBarnes Personally, I think single IP addresses are clearer, provided we update the help text for that field to to something like

> Enter the IP address ranges, one per line, that are allowed to view objects. Separate the low and high ends of each range with a colon, e.g. 111.111.111.111:222.222.222.222. Asterisks are not allowed. Single IP addresses are also allowed, each on its own line.

But it doesn't really matter to me, as long as users know what to do if they want to allow access from single IPs.

Comment (MarcusBarnes) [reply plugins/islandora_context_reaction_restrict_by_ip.inc:114]: That sounds good.  A separate issue to track work on allowing single IPs would be appropriate.  Thank you.

Comment (mjordan) [reply plugins/islandora_context_reaction_restrict_by_ip.inc:114]: OK,I'll close #30 and open a new one for single IPs.
